### DISCUSSION TITLE
how to resend verification email?

### DISCUSSION THREAD

**Discussion by flixyudh** [invertase/react-native-firebase#6174] [Q&A]
I'm using email and password for my login apps, and it's required to validate email.

here's my code to execute send email

```js
const userCredential = await auth().createUserWithEmailAndPassword(
  dataset.email,
  dataset.password,
);
const {email, password, ...anotherDataset} = dataset;
userCredential.user.updateProfile({displayName: dataset.name});
userCredential.user.sendEmailVerification();
auth().signOut(); //force logout to let user not going to Home, handled by onAuthStateChanged
```

and on my login `onPress` i used this to verify is an email already verified or not

```js
const userCredential = await auth().signInWithEmailAndPassword(
  email,
  password,
);
!userCredential?.user?.emailVerified &&
  FlixToast.show('Email not verify', {
    status: 'danger',
    type: 'snackbar',
    label: 'Resend',
    duration: 1500,
    onPress: () => userCredential.user.sendEmailVerification(),
  });
!userCredential?.user?.emailVerified && (await auth().signOut());
```

```userCredential.user.sendEmailVerification()``` seems not resend verification email, so is there any function to handle this?

**Comment by mikehardy** [comment]
Unsure what's happening but I have definitely re-sent verification emails.
Here's some sample code that may be interesting, the project itself runs out of the box so can at minimum be used as basis for a reproduction and is easy for you to check + experiment
It was working last time I ran through it, a few weeks back
https://github.com/invertase/react-native-firebase-authentication-example/

https://github.com/invertase/react-native-firebase-authentication-example/blob/68bd37f8c219f892655b40c73e9818370f368b82/template/src/app/signed-in/Settings.tsx#L164-L166

**Reply by flixyudh** [reply]
Thank you for your prompt reply @mikehardy .

I will give it a try
